enable tpm

Ham22 opened the pull request.

Ham22 committed:
kernel: add tpm support

This adds kernel support for tpm's providing:
* kmod-tpm - tpm framework
* kmod-tpm-i2c-infineon - infineon i2c 1.2 spec tpm
* kmod-random-tpm - tpm backed rng

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>

abhijit-mahajani-imgtec commented on package/kernel/linux/modules/tpm.mk:
```diff
@@ -0,0 +1,36 @@
+TPM_MENU:=TPM
+
+define KernelPackage/tpm
```
kernel: add tpm support -> 
It would be good to mention addition of "kmod-tpm support" in the commit message.

Ham22 replied:
OK, shall do. Note I have added a TPM menu as I am trying to follow roughly the same sections as the kernel uses and the SSL menu certainly didn't seem like the right place. Any suggestions or are we OK with this?

abhijit-mahajani-imgtec commented on target/linux/pistachio/dts/pistachio_marduk_common.dtsi:
```diff
@@ -280,6 +280,11 @@
 &i2c2 {
 	status = "okay";
 	clock-frequency = <400000>;
+	tpm_i2c_slb9645@20 {
```
would be good to have dts change in a separate commit so that we can mention explicitly

pistachio_marduk_common.dtsi: add tpm config or something like that
Mostly we have been following to mention "dts" or "creator-platform-default configs" filenames/module names in the commit messages.

Ham22 committed:
pistachio_marduk_common.dtsi: add infineon tpm

Add the tpm to the device tree which is common to all marduk
boards.

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>

Ham22 committed:
marduk: target.mk: enable tpm by default

All marduk variants have the same tpm so enable it on the target by
default along with some useful tools.

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>

abhijit-mahajani-imgtec commented:
Approved

abhijit-mahajani-imgtec commented on target/linux/pistachio/marduk/target.mk:
```diff
@@ -28,6 +28,7 @@ BOARDNAME:=marduk
 DEFAULT_PACKAGES+=kmod-leds-gpio kmod-ledtrig-heartbeat kmod-i2c-core i2c-tools \
                   kmod-sound-pistachio-soc alsa-lib alsa-utils alsa-utils-tests \
                   kmod-uccp420wlan kmod-cfg80211 fping iw hostapd wpa-cli wpa-supplicant \
+                  kmod-tpm-i2c-infineon kmod-random-tpm \
```
why shouldn't we enable tpm-tools too by default?

Ham22 replied:
Because gnutls package (which we have enabled) currently breaks the build as soon as it notices that trousers is available due to a missing reference 'tpmtool.c:(.text.startup+0x60c): undefined reference to pkcs11_common'.

Need to push fixes for that first. (Or we could enable GNUTLS_PKCS11 as a workaround for the bug)

Ham22 replied:
The fix was accepted upstream https://github.com/openwrt/packages/pull/3174/ so I shall add the default packages then this should be good to go.

abhijit-mahajani-imgtec commented:
test with cc2520 profile

abhijit-mahajani-imgtec commented:
test with jenkins - dev

nikhil-zinjurde-imgtec commented:
test with jenkins - dev

nikhil-zinjurde-imgtec commented:
test with jenkins - dev

nikhil-zinjurde-imgtec commented:
test with jenkins - dev

nikhil-zinjurde-imgtec commented:
test with ca8210 profile
